Server panic when using wgsd-client

Corefile:
```
.:5353 {
        debug
        wgsd japuba.net.br wg0 {
            self
        }
}
```

Server output:
```
./coredns
.:5353
CoreDNS-1.8.0
linux/arm, go1.15.6, 
[DEBUG] plugin/wgsd: received query for: mql7ok6ufp76w6t3xf22vvkygx4x22gdka4ayhn76poqtkplxm3q====._wireguard._udp. type: SRV
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x119c0e4]

goroutine 58 [running]:
github.com/jwhited/wgsd.getHostRR(0x420c960, 0x57, 0x0, 0x1, 0x458e101)
        /home/alarm/downloads/wgsd/wgsd.go:232 +0x14
github.com/jwhited/wgsd.handleSRV(0x422f5e0, 0x1739fa8, 0x411ff50, 0x412ac10, 0xe, 0x0, 0x420c960, 0x57, 0x0, 0x0, ...)
        /home/alarm/downloads/wgsd/wgsd.go:106 +0x198
github.com/jwhited/wgsd.(*WGSD).ServeDNS(0x44c6760, 0x1733540, 0x42ea360, 0x1739fa8, 0x411ff50, 0x422f5e0, 0x1b710, 0x411ff40, 0x10)
        /home/alarm/downloads/wgsd/wgsd.go:227 +0x2c8
github.com/coredns/coredns/core/dnsserver.(*Server).ServeDNS(0x44cb080, 0x1733540, 0x42ea360, 0x1739fa8, 0x411ff50, 0x422f5e0)
        /home/alarm/go/pkg/mod/github.com/coredns/coredns@v1.8.0/core/dnsserver/server.go:289 +0x3b4
github.com/coredns/coredns/core/dnsserver.(*Server).ServePacket.func1(0x173ae18, 0x4480960, 0x422f5e0)
        /home/alarm/go/pkg/mod/github.com/coredns/coredns@v1.8.0/core/dnsserver/server.go:126 +0x80
github.com/miekg/dns.HandlerFunc.ServeDNS(0x41ea030, 0x173ae18, 0x4480960, 0x422f5e0)
        /home/alarm/go/pkg/mod/github.com/miekg/dns@v1.1.34/server.go:37 +0x34
github.com/miekg/dns.(*Server).serveDNS(0x41f20a0, 0x40e4200, 0x68, 0x200, 0x4480960)
        /home/alarm/go/pkg/mod/github.com/miekg/dns@v1.1.34/server.go:617 +0x298
github.com/miekg/dns.(*Server).serveUDPPacket(0x41f20a0, 0x41e8060, 0x40e4200, 0x68, 0x200, 0x44c8368, 0x411ff10)
        /home/alarm/go/pkg/mod/github.com/miekg/dns@v1.1.34/server.go:557 +0x98
created by github.com/miekg/dns.(*Server).serveUDP
        /home/alarm/go/pkg/mod/github.com/miekg/dns@v1.1.34/server.go:487 +0x1fc
```

Client output:
```
./wgsd-client -device wg0 -dns 10.34.33.2:5353 -zone japuba.net.br
2021/01/06 22:52:28 [ZBf3K9Qr/+t6e7l1qtVYNfl9aMNQOAwdv/PdCanruzc=] failed to lookup SRV: read udp 10.34.33.10:45163->10.34.33.2:5353: i/o timeout
2021/01/06 22:52:28 [RP2teEPOF6La6jkqUGt/zXeiYRLdwYnbNNdqV7898RE=] failed to lookup SRV: read udp 10.34.33.10:41977->10.34.33.2:5353: read: connection refused
2021/01/06 22:52:29 [h4E4dbJoTSwfQGtudK+qmSUjd0hhR3zPTYs8tDbW21Q=] failed to lookup SRV: read udp 10.34.33.10:47761->10.34.33.2:5353: read: connection refused
2021/01/06 22:52:29 [OxmA/ZUQlTWrRSV+2QLyPb9AS6YkeB3ZSYGLcH8zBl4=] failed to lookup SRV: read udp 10.34.33.10:55267->10.34.33.2:5353: read: connection refused
2021/01/06 22:52:29 [EYu9sqfU/7ebk/dYWQ4y4UxYve/PGPwNSdOipVpIqEY=] failed to lookup SRV: read udp 10.34.33.10:46015->10.34.33.2:5353: read: connection refused
```

--- a/wgsd.go
+++ b/wgsd.go
@@ -103,6 +103,9 @@ func handleSRV(state request.Request, peers []wgtypes.Peer) (int, error) {
 		if strings.EqualFold(
 			base32.StdEncoding.EncodeToString(peer.PublicKey[:]), pubKey) {
 			endpoint := peer.Endpoint
+			if endpoint == nil {
+				return nxDomain(state)
+			}
 			hostRR := getHostRR(state.Name(), endpoint)
 			if hostRR == nil {
 				return nxDomain(state)
@@ -137,6 +140,9 @@ func handleHostOrTXT(state request.Request, peers []wgtypes.Peer) (int, error) {
 		if strings.EqualFold(
 			base32.StdEncoding.EncodeToString(peer.PublicKey[:]), pubKey) {
 			endpoint := peer.Endpoint
+			if endpoint == nil {
+				return nxDomain(state)
+			}
 			if state.QType() == dns.TypeA || state.QType() == dns.TypeAAAA {
 				hostRR := getHostRR(state.Name(), endpoint)
 				if hostRR == nil {

--- a/wgsd_test.go
+++ b/wgsd_test.go
@@ -72,6 +72,15 @@ func TestWGSD(t *testing.T) {
 	}
 	peer2b32 := strings.ToLower(base32.StdEncoding.EncodeToString(peer2.PublicKey[:]))
 	peer2b64 := base64.StdEncoding.EncodeToString(peer2.PublicKey[:])
+	key3 := [32]byte{}
+	key3[0] = 3
+	peer3Allowed, _ := constructAllowedIPs(t, []string{"10.0.0.5/32", "10.0.0.6/32"})
+	peer3 := wgtypes.Peer{
+		Endpoint:   nil,
+		PublicKey:  key3,
+		AllowedIPs: peer3Allowed,
+	}
+	peer3b32 := strings.ToLower(base32.StdEncoding.EncodeToString(peer3.PublicKey[:]))
 	p := &WGSD{
 		Next: test.ErrorHandler(),
 		Zones: Zones{
@@ -91,7 +100,7 @@ func TestWGSD(t *testing.T) {
 					Name:       "wg0",
 					PublicKey:  selfKey,
 					ListenPort: 51820,
-					Peers:      []wgtypes.Peer{peer1, peer2},
+					Peers:      []wgtypes.Peer{peer1, peer2, peer3},
 				},
 			},
 		},
@@ -204,6 +213,46 @@ func TestWGSD(t *testing.T) {
 			Qname: "servfail.notexample.com.",
 			Qtype: dns.TypeAAAA,
 			Rcode: dns.RcodeServerFailure,
+		},
+		{
+			Qname: fmt.Sprintf("%s._wireguard._udp.example.com.", peer3b32),
+			Qtype: dns.TypeSRV,
+			Rcode: dns.RcodeNameError,
+			Ns: []dns.RR{
+				test.SOA(soa("example.com.").String()),
+			},
+			Answer: []dns.RR{},
+			Extra:  []dns.RR{},
+		},
+		{
+			Qname: fmt.Sprintf("%s._wireguard._udp.example.com.", peer3b32),
+			Qtype: dns.TypeA,
+			Rcode: dns.RcodeNameError,
+			Ns: []dns.RR{
+				test.SOA(soa("example.com.").String()),
+			},
+			Answer: []dns.RR{},
+			Extra:  []dns.RR{},
+		},
+		{
+			Qname: fmt.Sprintf("%s._wireguard._udp.example.com.", peer3b32),
+			Qtype: dns.TypeAAAA,
+			Rcode: dns.RcodeNameError,
+			Ns: []dns.RR{
+				test.SOA(soa("example.com.").String()),
+			},
+			Answer: []dns.RR{},
+			Extra:  []dns.RR{},
+		},
+		{
+			Qname: fmt.Sprintf("%s._wireguard._udp.example.com.", peer3b32),
+			Qtype: dns.TypeTXT,
+			Rcode: dns.RcodeNameError,
+			Ns: []dns.RR{
+				test.SOA(soa("example.com.").String()),
+			},
+			Answer: []dns.RR{},
+			Extra:  []dns.RR{},
 		},
 	}
 	for _, tc := range testCases {

